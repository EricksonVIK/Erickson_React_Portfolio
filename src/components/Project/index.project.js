import React, { useState } from "react";
import noteTaker from "../../assets/images/noteTaker2.png";
import weatherApp from "../../assets/images/weatherApp.png";
import textEditor from "../../assets/images/JATE_screenshot.png";
import YodaIM from "../../assets/images/yodaIM.png";
import github from "../../assets/images/GitHub-icon.png";

function Project() {
  // code in project array -- need to add more
  const projects = [
    {
      src: noteTaker,
      title: "Note Taker",
      tools: "Express.js, Node.js",
      deploy: "https://whispering-escarpment-89419.herokuapp.com/",
      github: "https://github.com/EricksonVIK/Note_Taker",
    },
    {
      src: weatherApp,
      title: "Weather Finder",
      tools: "HTML, CSS, JavaScript",
      deploy: "https://ericksonvik.github.io/Weather-Finder/",
      github: "https://github.com/EricksonVIK/Weather-Finder",
    },
    {
      src: textEditor,
      title: "Text Editor",
      tools: "Node.js, Express.js, WebPack, IndexedDB",
      deploy: "https://serene-anchorage-57681.herokuapp.com/",
      github: "https://github.com/EricksonVIK/text-editor",
    },
    {
      src: YodaIM,
      title: "GifYoda",
      tools: "Materialize CSS, API's",
      deploy: "https://ericksonvik.github.io/YodaIM/",
      github: "https://github.com/EricksonVIK/YodaIM",
    },
  ];

  const [projectCard, setProjectCard] = useState(projects[0]);
  const [hover, setHover] = useState(false);

  //jsx
  return (
    <section>
      <h2>Portfolio</h2>
      <div className="portfolio-section">
        {projects.map((project) => (
          <div
            key={project.title}
            className="container"
            onMouseEnter={() => {
              setHover(true);
              setProjectCard(project);
            }}
            onMouseLeave={() => {
              setHover(false);
              setProjectCard(project);
            }}
          >
            {hover && projectCard.title === project.title ? (
              <>
                <img src={`${project.src}`} alt="" className="img"></img>
                <div className="project-details">
                  <span className="flex-row">
                    <a
                      href={`${project.deploy}`}
                      target="blank"
                      className="deployed"
                    >
                      {project.title}
                    </a>
                    <a href={`${project.github}`} target="blank">
                      <img
                        className="github-icon"
                        alt="github icon"
                        src={`${github}`}
                      />
                    </a>
                  </span>
                  <p className="tools">{project.tools}</p>
                </div>
              </>
            ) : (
              <img
                src={`${project.src}`}
                alt={`${project.title}`}
                className="image"
              ></img>
            )}
          </div>
        ))}
      </div>
    </section>
  );
}

export default Project;
