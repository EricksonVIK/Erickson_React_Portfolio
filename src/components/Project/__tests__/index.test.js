import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Project from "../index.project";

afterEach(cleanup);

describe("Project component", () => {
  // baseline test
  it("renders", () => {
    render(<Project />);
  });
  // snapshot test
  it("matches snapshot", () => {
    const { asFragment } = render(<Project />);
    // assert value comparison
    expect(asFragment()).toMatchSnapshot();
  });
});
