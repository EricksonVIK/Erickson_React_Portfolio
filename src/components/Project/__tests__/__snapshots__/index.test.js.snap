// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project component matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <h2>
      Portfolio
    </h2>
    <div
      class="portfolio-section"
    >
      <div
        class="container"
      >
        <img
          alt="Note Taker"
          class="image"
          src="noteTaker2.png"
        />
      </div>
      <div
        class="container"
      >
        <img
          alt="Weather Finder"
          class="image"
          src="weatherApp.png"
        />
      </div>
      <div
        class="container"
      >
        <img
          alt="Text Editor"
          class="image"
          src="JATE_screenshot.png"
        />
      </div>
      <div
        class="container"
      >
        <img
          alt="GifYoda"
          class="image"
          src="yodaIM.png"
        />
      </div>
    </div>
  </section>
</DocumentFragment>
`;
